test/integration/pandas_test.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import pyexasol
+from inspect import cleandoc
+import pandas as pd
+from pandas import testing as pd_testing
+from operator import itemgetter
+
+
+@pytest.fixture
+def connection_with_compression(dsn, user, password, schema):
+    with pyexasol.connect(
+        dsn=dsn,
+        user=user,
+        password=password,
+        schema=schema,
+        compression=True,
+    ) as connection:
+        yield connection
+
+
+@pytest.fixture
+def empty_table(connection):
+    name = "USER_NAMES"
+    ddl = cleandoc(
+        f"""
+        CREATE OR REPLACE TABLE {name}
+        (
+            FIRST_NAME VARCHAR(200),
+            LAST_NAME VARCHAR(200)
+        );
+        """
+    )
+    connection.execute(ddl)
+    connection.commit()
+
+    yield name
+
+    ddl = f"DROP TABLE IF EXISTS {name};"
+    connection.execute(ddl)
+    connection.commit()
+
+
+@pytest.fixture
+def names(faker):
+    yield tuple(
+        {"FIRST_NAME": faker.first_name(), "LAST_NAME": faker.last_name()}
+        for _ in range(0, 10)
+    )
+
+
+@pytest.fixture
+def table(connection, empty_table, names):
+    insert = "INSERT INTO {table} VALUES({{FIRST_NAME}}, {{LAST_NAME}});"
+    for name in names:
+        stmt = insert.format(table=empty_table)
+        connection.execute(stmt, name)
+
+    yield empty_table, names
+
+
+@pytest.mark.parametrize(
+    "connection", ["connection", "connection_with_compression"], indirect=True
+)
+@pytest.mark.pandas
+def test_export_table_to_pandas(connection, table):
+    table_name, values = table
+
+    expected = pd.DataFrame.from_records(values)
+    actual = connection.export_to_pandas(table_name)
+
+    assert actual.equals(expected)
```

Comment:
Maybe it's better to use the `pandas.testing.assert_frame_equal`. Then you will have more options. For example you can disregard the row order, which might be useful.

Reply:
Hi @ahsimb,

Good point, I must admit I had it in before but for me this version seemed to be simpler to use :sweat_smile: likely because I am not so familiar with pandas and it wasn't immediately obvious which settings to use, because the default did not seem to be sufficient in my case.